### PR TITLE
feat: v-on modifiers support native options and keyboards

### PR DESCRIPTION
**Pull request by baiwusanyu-c**
close: #26

### CODE REVIEW

**Comment by baiwusanyu-c** [comment]
There are currently some inconsistent use cases with `vue/core`
* Caused by `isStaticExp(key)`
```
<input @keyup.right="submit" /> 
<input @keyup.right.self="submit" />
<input @keyup.right.enter="submit" />
```
* Need support for `@foo`
```
<form @submit.prevent></form> 
```

File: packages/compiler-vapor/src/generate.ts [hunk 1]
```diff
@@ -443,3 +434,36 @@ function genExpression(
 
   push(content, NewlineType.None, exp.loc, name)
 }
+
+function genEventOperation(oper: SetEventIRNode, context: CodegenContext) {
+  const { vaporHelper, push, pushWithNewline } = context
+
+  pushWithNewline(`${vaporHelper('on')}(n${oper.element}, `)
+  genExpression(`"${oper.name}"`, context)
```
[the remark this answers is not in the document]
Comment (baiwusanyu-c) [reply packages/compiler-vapor/src/generate.ts]: like it ?
```
push('"')
genExpression(oper.name, context)
push('"')
```

File: packages/compiler-vapor/src/transforms/vOn.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  if (!arg) {
+    // TODO support v-on="{}"
+    return
+  } else if (exp === undefined) {
+    // TODO: support @foo
+    return
+  }
+
+  const { keyModifiers, nonKeyModifiers, eventOptionModifiers } =
+    resolveModifiers(exp, modifiers, null, loc)
```
Comment (sxzz) [inline packages/compiler-vapor/src/transforms/vOn.ts:42]: Should it be `arg` instead of `exp`?

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
Refactored a lot

**Review by sxzz — APPROVED** [review]

**Pull request merged**